src/utils/handleOptions.js

```diff
@@ -0,0 +1,176 @@
+const errorMessages = {
+  type: (modelName, keyName, userType, requiredType, defaultValue) =>
+    `🟪ml5.js warns: The '${keyName}' option for ${modelName} has to be set to a ${requiredType}, but it is being set to a ${userType} instead.\n\nml5.js is using default value of '${defaultValue}'.`,
+  enum: (modelName, keyName, userValue, enums, defaultValue) =>
+    `🟪ml5.js warns: The '${keyName}' option for ${modelName} has to be set to ${renderArray(
+      enums
+    )}, but it is being set to '${userValue}' instead.\n\nml5.js is using default value of '${defaultValue}'.`,
+  numberRange: (modelName, keyName, userValue, min, max, defaultValue) =>
+    `🟪ml5.js warns: The '${keyName}' option for ${modelName} has to be set to a number between ${min} and ${max}, but it is being set to '${userValue}' instead.\n\nml5.js is using default value of '${defaultValue}'.`,
+  numberInteger: (modelName, keyName, userValue, defaultValue) =>
+    `🟪ml5.js warns: The '${keyName}' option for ${modelName} has to be set to an integer, but it is being set to the float value '${userValue}' instead.\n\nml5.js is using the default value of ${defaultValue}.`,
+};
+
+/**
+ * Transforms an array into a human readable string, wrapping the elements with quotation, splitting them with a comma and the last element with "or".
+ * @param {array} array
+ * @returns {string} the string representation of the array
+ */
+function renderArray(array) {
+  if (array.length === 1) {
+    return `'${array[0]}'`;
+  } else if (array.length === 2) {
+    return `'${array[0]}' or '${array[1]}'`;
+  } else {
+    const last = array.pop();
+    return `'${array.join("', '")}', or '${last}'`;
+  }
+}
+
+/**
+ * Evaluates the value of a moldObject property.
+ * If the value is a function, it is called with the filtered options object as the its parameter.
+ * Otherwise, the value is returned as is.
+ * @param {object} filteredObject - the current filtered options object
+ * @param {any} value - the value to evaluate
+ * @returns {any} the evaluated value
+ */
+function evaluate(filteredObject, value) {
+  if (typeof value === "function") {
+    return value(filteredObject);
+  } else {
+    return value;
+  }
+}
+
+/**
+ * Checks if a value is within the enumArray.
+ * If isCaseInsensitive is true, the comparison is case insensitive.
+ *
+ * @param {any} value - the value to check
+ * @param {Array} enumArray - the array of enum values
+ * @param {boolean} isCaseInsensitive - whether the comparison is case insensitive
+ * @returns {any | undefined} the enum value if the user value is within the enumArray, otherwise undefined
+ */
+function checkEnum(value, enumArray, isCaseInsensitive) {
+  if (isCaseInsensitive && typeof value === "string") {
+    const matchingValue = enumArray.find(
+      (enumValue) => enumValue.toLowerCase() === value.toLowerCase()
+    );
```

Comment:
I'm thinking about numeric enums (ie. `values: [2, 4, 8]`) and if they would break anything.  I think that what you have here is fine because the scenario that throws an error relies on there being nonsense in the `moldObject`, which we control.  A bad user value alone can't break it.  The nonsense way to break it would be using case-insensitive checking with a numeric `enumArray`, and even then it's only an error if the user provides a string: `checkEnum("a", [2, 4, 8], true)`.